persistence-typed/src/main/scala/org/apache/pekko/persistence/typed/state/internal/DurableStateStoreException.scala

```diff
@@ -13,18 +13,40 @@
 
 package org.apache.pekko.persistence.typed.state.internal
 
+import scala.util.control.NoStackTrace
+
 import org.apache.pekko
-import pekko.annotation.InternalApi
 import pekko.persistence.typed.PersistenceId
 
 /**
- * INTERNAL API
- *
- * Used for store failures. Private to pekko as only internal supervision strategies should use it.
+ * Base class for exceptions thrown by the Durable State implementations.
+ * Public since v1.1.0.
  */
-@InternalApi
-final private[pekko] class DurableStateStoreException(msg: String, cause: Throwable)
+class DurableStateStoreException(msg: String, cause: Throwable)
```

Comment:
I think we should keep the 
`@InternalApi` at least.

Reply:
I reworked it again. Made DurableStateStoreException InteranlApi again but moved the new classes to a new package. The new classes are not internal, writers of persistence implementations like https://github.com/SwissBorg/pekko-persistence-postgres may want to make the same chnages that we are going to make to https://github.com/apache/pekko-persistence-jdbc and other projects to have the deleteObject function fail with one of these new exceptions if no DB row is deleted.